Erin/studio assessment controller

erinysong opened the pull request:
## Studio Assessment Controller

Created Studio Assessment Controller that performs CRUD functionality (similar to casenotes and paperwork)
Also changed some fields to the Studio Assessment Table but if there are any issues with that feel free to let me know

### Related PRs

Optional - any related PRs you're waiting on, or PRs that will conflict, etc

### Migrations

Optional - if you added anything to the database through migration(s)


### Screenshots

Optional but would be nice for final presentation :)


CC: @didvi

julianrkung commented on app/controllers/api/studio_assessments_controller.rb:
```diff
@@ -0,0 +1,71 @@
+
+    private
+
+    def set_studio_assessment
```
Divi and I talked about not doing authorization in set_studio_assessment and instead moving the authorization to each of the individual CRUD functions (so adding an authorize call to show, update, and destroy). I think that makes more sense? Would like your input! If we decide to do authorization in set_studio_assessments I can update the studio_asessment_policy pundit though. :)

julianrkung commented on db/migrate/20200307003647_create_studio_assessments.rb:
```diff
@@ -0,0 +1,29 @@
+      t.string :problemsolving_comment
+      t.integer :problemsolvingalt_score
+      t.string :problemsolvingalt_comment
+      t.boolean :passed_capstone
```
maybe `capstone_passed` to keep consistent w/above styling. If we change this we need to change it in the api controller and views and others also!

julianrkung commented on db/schema.rb:
```diff
@@ -93,6 +93,34 @@
+    t.string "problemsolving_comment"
+    t.integer "problemsolvingalt_score"
+    t.string "problemsolvingalt_comment"
+    t.boolean "passed_capstone"
```
`capstone_passed`

erinysong commented on app/assets/stylesheets/scaffolds.scss:
```diff
@@ -0,0 +1,84 @@
+body {
```
think divi ended up deleting this file and it seems to be gone after i pulled from master so thanks for pointing this out! def should have deleted all the unnecessary files that scaffold generated

erinysong commented on app/controllers/api/studio_assessments_controller.rb:
```diff
@@ -0,0 +1,71 @@
+
+    def studio_assessment_params
+        case_notes_param = params.require(:studio_assessment).permit(:participant_id, 
+                                                                    :staff_id,
```
yeah you're right I completely overlooked deleting that field after I added the merge line - just fixed it, thanks for catching that :))

erinysong commented:
addressed all the comments made, feel free to let me know if there was anything I missed but I made the following changes:

removed unnecessary extra staff_id field since we have a line that merges it from the staff table anyway
moved authorization to the individuals CRUD functions in studio assessments controller
changed passed_capstone field in studio assessments db to capstone_passed for consistence

erinysong commented:
just made the changes that you requested above, thanks divi! :)
